code/clue.js

```diff
@@ -1,59 +1,347 @@
+
+  mystery.killer = randomSelector(suspects) // This will randomly select a killer from the suspects. And add that to the mystery object.
+
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+  // This will change what information about the killer is displayed
   document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerImage').src = mystery.killer.image
+  document.getElementById('killerOccupation').innerHTML = `${mystery.killer.occupation}`
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerDescription').innerHTML = `${mystery.killer.description}`
+
+  // SHUFFLE FAVOURITE WEAPON
+  const shuffleFavouriteWeapon = () => {
+    mystery.favouriteWeapon = randomSelector(weapons)
+    document.getElementById('favouriteWeapon').innerHTML = `Favourite Weapon: ${mystery.favouriteWeapon.name}`
+  }
+  shuffleFavouriteWeapon()
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+
+
+// RANDOM WEAPON FUNCTION
+
+// FUNCTION TO DELAY PICK WEAPON FUNCTION
+const delayPickWeapon = () => {
+  document.getElementById('loaderWeapon').style.display = 'block'
+  setTimeout(pickWeapon, 1000)
+}
+document.getElementById('weaponCard').addEventListener('click', delayPickWeapon);
+
+// PICK WEAPON FUNCTION
+const pickWeapon = () => { 
+  document.getElementById('loaderWeapon').style.display = 'none'
+  // document.getElementById('weaponCard').removeEventListener('click', delayPickWeapon)
+  
+  mystery.weapon = randomSelector(weapons)
+
+  document.getElementById('weaponCard').style.background = '#FFFFFF';
+  document.getElementById('weaponImage').src = mystery.weapon.image;
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`;
+}
+
+
+// RANDOM ROOM FUNCTION
+
+// FUNCTION TO DELAY PICK ROOM FUNCTION
+const delayPickRoom = () => {
+  document.getElementById('loaderRoom').style.display = 'block'
+  setTimeout(pickRoom, 1000)
+}
+document.getElementById('roomCard').addEventListener('click', delayPickRoom)
+
+// PICK ROOM FUNCTION
+const pickRoom = () => { 
+  document.getElementById('loaderRoom').style.display = 'none'
+  // document.getElementById('roomCard').removeEventListener('click', delayPickRoom)
+  
+  mystery.room = randomSelector(rooms)
+
+  document.getElementById('roomCard').style.background = '#FFFFFF';
+  document.getElementById('roomName').innerHTML = `${mystery.room.name}`;
```

Comment:
You seem to add semicolons only from time to time - we have to be consequent about that ;)